alicloud/resource_alicloud_gpdb_elastic_instance.go

```diff
@@ -0,0 +1,336 @@
+		if err != nil {
+			return WrapError(err)
+		}
+		request["VPCId"] = vsw["VpcId"]
+		request["VSwitchId"] = vswitchId
+		if v, ok := request["ZoneId"].(string); !ok || v == "" {
+			request["ZoneId"] = vsw["ZoneId"]
+		}
+	}
+	request["ClientToken"] = buildClientToken("CreateECSDBInstance")
+	wait := incrementalWait(3*time.Second, 3*time.Second)
+	runtime := util.RuntimeOptions{}
+	runtime.SetAutoretry(true)
+	err = resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		response, err = conn.DoRequest(StringPointer(action), nil, StringPointer("POST"), StringPointer("2016-05-03"), StringPointer("AK"), nil, request, &runtime)
+		if err != nil {
+			if NeedRetry(err) {
+				wait()
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		addDebug(action, response, request)
+		return nil
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_gpdb_elastic_instance", action, AlibabaCloudSdkGoERROR)
+	}
+
+	d.SetId(fmt.Sprint(response["DBInstanceId"]))
+	stateConf := BuildStateConf([]string{}, []string{"Running"}, d.Timeout(schema.TimeoutCreate), 10*time.Minute, gpdbService.GpdbElasticInstanceStateRefreshFunc(d.Id(), []string{}))
+	if _, err := stateConf.WaitForState(); err != nil {
+		return WrapErrorf(err, IdMsg, d.Id())
+	}
+
+	return resourceAlicloudGpdbElasticInstanceUpdate(d, meta)
+}
+
+func resourceAlicloudGpdbElasticInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	gpdbService := GpdbService{client}
+	instance, err := gpdbService.DescribeGpdbElasticInstance(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.Set("engine", instance["Engine"])
+	d.Set("engine_version", instance["EngineVersion"])
+	d.Set("storage_type", instance["StorageType"])
+	d.Set("seg_node_num", instance["SegNodeNum"])
+	d.Set("storage_size", instance["StorageSize"])
+	d.Set("payment_type", convertGpdbInstancePaymentTypeResponse(instance["PayType"].(string)))
+	d.Set("instance_spec", convertDBInstanceClassToInstanceSpec(instance["DBInstanceClass"].(string)))
+	d.Set("status", instance["DBInstanceStatus"])
+	d.Set("description", instance["DBInstanceDescription"])
+	d.Set("instance_network_type", instance["InstanceNetworkType"])
+	d.Set("vswitch_id", instance["VSwitchId"])
+	d.Set("zone_id", instance["ZoneId"])
+	d.Set("connection_string", instance["ConnectionString"])
+	securityIps, err := gpdbService.DescribeGpdbSecurityIps(d.Id())
+	if err != nil {
+		return WrapError(err)
+	}
+	d.Set("security_ip_list", securityIps)
+	return nil
+}
+
+func resourceAlicloudGpdbElasticInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	gpdbService := GpdbService{client}
+	d.Partial(true)
+	conn, err := client.NewGpdbClient()
+	if err != nil {
+		return WrapError(err)
+	}
+	if d.HasChange("description") {
```

Comment:
The description has been set in the Create API and there should check the ` d.IsNewResource() `